engine/src/main/java/io/camunda/zeebe/engine/processing/common/CatchEventBehavior.java

```diff
@@ -311,13 +287,9 @@ public void subscribeToTimerEvent(
         .setTargetElementId(handlerNodeId)
         .setProcessDefinitionKey(processDefinitionKey);
 
-    sideEffects.add(
-        () -> {
-          /* timerChecker implements onRecovered to recover from restart, so no need to schedule
-          this in TimerCreatedApplier.*/
-          timerChecker.scheduleTimer(dueDate);
-          return true;
-        });
+    /* timerChecker implements onRecovered to recover from restart, so no need to schedule
+    this in TimerCreatedApplier.*/
+    timerChecker.scheduleTimer(dueDate);
```

Comment:
>  The timer task of the due checker can be submitted between the command processing and the database commit.

Actually it can't since the one is in the fast lane and the other is a submitted job, but anyway I get what you mean and that is an implementation detail. Wondering what would be the solution here. I think this can be solved differently, with the proposed scheduling service we can introduce (part of the proposal)

Reply:
> I think this can be solved differently, with the proposed scheduling service we can introduce (part of the proposal)